src/main/java/org/wmn4j/notation/elements/Marking.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Distributed under the MIT license (see LICENSE.txt or https://opensource.org/licenses/MIT).
+ */
+package org.wmn4j.notation.elements;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Represents an articulation marking that spans across multiple notes, such as
+ * a slur or a glissando. The articulations are represented using connections
+ * between adjacent notes. For example, if three notes are affected the same
+ * slur, the first will be connected to the second note and the second note will
```

Comment:
This should probably be "affected by the same slur"?

Reply:
Fixed the typo. Also fixed the case where note builders can be connected to each other through ties to form a circular dependency and an infinite build loop.